validator/client/src/main/java/tech/pegasys/teku/validator/client/ActiveKeyManager.java

```diff
@@ -130,6 +131,33 @@ public synchronized DeleteKeysResponse deleteValidators(
     return new DeleteKeysResponse(deletionResults, exportedData);
   }
 
+  @Override
+  public DeleteRemoteKeysResponse deleteExternalValidators(List<BLSPublicKey> validators) {
+    final List<DeleteKeyResult> deletionResults = new ArrayList<>();
+    for (final BLSPublicKey publicKey : validators) {
+      Optional<Validator> maybeValidator =
+          validatorLoader.getOwnedValidators().getValidator(publicKey);
+
+      // read-only check in a non-destructive manner
+      if (maybeValidator.isPresent() && maybeValidator.get().isReadOnly()) {
+        deletionResults.add(DeleteKeyResult.error("Cannot remove read-only validator"));
+        continue;
+      }
+      // delete validator from owned validators list
```

Comment:
after this, check the validator isn't local. should probably check on the local side that it *is* local... otherwise the call would fail anyway below

Reply:
im not 100% sure we can get into that position, did your investigations show that kind of behaviour?